lib/src/main/kotlin/sdk/kotlin/Library.kt

```diff
@@ -131,16 +184,35 @@ public class VerifiablePresentation {
             // TODO change to be more than one VC
             val vp: VerifiablePresentationType = VerifiablePresentationType.builder()
                 .verifiableCredential(createVpOptions.verifiableCredentials[0])
-                .build();
+                .holder(URI.create(createVpOptions.holder))
+                .build()
 
-            return ToJwtConverter.toJwtVerifiablePresentation(vp).sign_Ed25519_EdDSA(signOptions.signerPrivateKey)
+            return ToJwtConverter.toJwtVerifiablePresentation(vp)
+                .sign_Ed25519_EdDSA(signOptions.signerPrivateKey.toOctetKeyPair(), signOptions.kid, false)
         }
 
-        public fun verify(publicKey: OctetKeyPair, vpJWT: String): Boolean {
-            require(!publicKey.isPrivate)
+        public fun verify(vpJWT: String, resolver: DIDResolver): Boolean {
+            val publicKeyJWK = issuerPublicJWK(vpJWT, resolver)
+            require(!publicKeyJWK.isPrivate)
             require(vpJWT.isNotEmpty())
-            // TODO: Have did resolution verification
-            return JwtVerifiablePresentation.fromCompactSerialization(vpJWT).verify_Ed25519_EdDSA(publicKey)
+            return JwtVerifiablePresentation.fromCompactSerialization(vpJWT)
+                .verify_Ed25519_EdDSA(publicKeyJWK.toOctetKeyPair())
         }
     }
 }
+
+private fun issuerPublicJWK(vcJWT: String, resolver: DIDResolver): JWK {
+    val signedJWT = SignedJWT.parse(vcJWT)
+    val jwsHeader = signedJWT.header
+    val kid = jwsHeader.keyID
+    val issuer: String = signedJWT.jwtClaimsSet.issuer
+
+    val result: ResolveDataModelResult = resolver.resolve(issuer, null)
+
+    val publicKeyJwkMap = result.didDocument.verificationMethods.first {
```

Comment:
will we only support the first for now? Is this how the ssi service does it?

Reply:
first returns the first element matching the predicate. See https://kotlinlang.org/api/latest/jvm/stdlib/kotlin.collections/first.html

In ssi-service, we do a similar search.